src/Mailgun/Api/Pagination.php

```diff
@@ -22,6 +23,51 @@
     abstract protected function safeDeserialize(ResponseInterface $response, $className);
 
     /**
+     * @return string
+     */
+    abstract protected function getPaginationBase();
+
+    /**
+     * @param PaginationResponse $response
+     *
+     * @return mixed|null
+     */
+    public function getPaginationNext(PaginationResponse $response)
```

Comment:
PaginationResponse is a Trait. You cannot use Traits like this. 
https://3v4l.org/K4mdT